src/solidity-frontend/solidity_language.cpp

```diff
@@ -9,8 +9,8 @@ CC_DIAGNOSTIC_POP()
 #include <solidity-frontend/solidity_language.h>
 #include <solidity-frontend/solidity_convert.h>
 #include <clang-c-frontend/clang_c_main.h>
-#include <clang-c-frontend/clang_c_adjust.h>
-#include <clang-c-frontend/clang_c_convert.h>
+#include <clang-cpp-frontend/clang_cpp_adjust.h>
```

Comment:
Why did we move from clang-c to clang-cpp?

Reply:
We should provide some explanation here.

Reply:
The general idea of the migration from C to C++ is that we treat each contract as a C++ **class**. Since clang-c did not support the feature of class or "construct a temporary object" or constructor call, therefore we change it to C++. The `mode` info of each symbol will be change from `C(0)` to `C++(1)`.

As for the reason for the code change here, it's because we need the adjustment in `clang_cpp_adjust_expr.cpp` which will first adjust the "temporary_object" expression (i.e. Base x = new Base()) and then feed it to `clang_c_adjust`. The `clang_c_adjust` cannot adjust this expression by itself.

I will add some documentation in the file to explain

Reply:
But some changes might not be necessary. I will take a look and do some testing to see what must be changed.